domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/caserole/CaseRoleEntityUniquenessValidatorImpl.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.caserole;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.SimpleValidationError;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationErrorMessageCreator;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseRoleEntity;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class CaseRoleEntityUniquenessValidatorImpl implements CaseRoleEntityValidator {
+
+    private static final String CREATOR = "[CREATOR]";
+
+    @Override
+    public ValidationResult validate(CaseRoleEntity caseRoleEntity,
+                                     CaseRoleEntityValidationContext caseRoleEntityValidationContext) {
+        final ValidationResult validationResult = new ValidationResult();
+        if (caseRoleEntity.getReference() != null && caseRoleEntity.getReference().equalsIgnoreCase(CREATOR)) {
```

Comment:
I reckon this one can be simplied as 

if (CREATOR.equalsIgnoreCase(caseRoleEntity.getReference()) {

as we know CREATER is definitely not null

Reply:
absolutely